### ISSUE TITLE
Install of older 0.3.3 broken

### ISSUE THREAD

**Issue by jvanns** [codedellemc/rexray#468]
Following this to ensure we maintain a version 0.3.3 (0.4 not working for us);

http://rexray.readthedocs.io/en/stable/user-guide/installation/#manual-installs

The output of;

curl -sSL https://dl.bintray.com/emccode/rexray/install | sh -s -- stable 0.3.3

Is actually;
dpkg-deb: error: `rexray-0.3.3-x86_64.deb' is not a debian format archive
dpkg: error processing archive rexray-0.3.3-x86_64.deb (--install):
 subprocess dpkg-deb --control returned error exit status 2
Errors were encountered while processing:
 rexray-0.3.3-x86_64.deb

Browsing the bintray hierarchy above there is indeed no .deb there for release 0.3.3

**Comment by kacole2** [comment]
@jvanns thanks for pointing this out. 

@akutz @clintonskitson verified that it's not location at this expected location, only tarballs. https://dl.bintray.com/emccode/rexray/stable/0.3.3/

**Comment by akutz** [comment]
Newer versions will retain RPMs, but no older. I'll make sure to upload the 0.3.3 RPMs manually in a bit

-- 
-a

**Comment by jvanns** [comment]
And Debian/Ubuntu packages?

**Comment by akutz** [comment]
Hi James,

It's the same process, so yes. I'm also out this afternoon, so someone else may need to handle it, but I am looking at it now at least.

**Comment by jvanns** [comment]
Great! Thanks :)

Jim

**Comment by kacole2** [comment]
@akutz i see the files have been uploaded but I believe the filename might be wrong.

```
[ec2-user@rhel72 ~]$ curl -sSL https://dl.bintray.com/emccode/rexray/install | sh -s stable 0.3.3
curl: (22) The requested URL returned error: 404 Not Found
error: skipping https://dl.bintray.com/emccode/rexray/stable/0.3.3/rexray-rexray-0.3.3-x86_64.rpm - transfer failed
```

**Comment by akutz** [comment]
Hi Kenny,

I'm indisposed, but check with Dre and Clint. There's another issue I plan on having resolved by EOD. They're aware of it.

-- 
-a

**Comment by kacole2** [comment]
@jvanns files have been renamed to work with the current script. it's a bandaid for now but will make sure to not break backwards compatibility for the next release. thank you. `.deb` should pull fine now.

thanks!

**Comment by kacole2** [comment]
@jvanns I'm working on getting a `.deb` file. Looks like we don't have a `.deb` laying around so I'm trying to build from source and hitting #472. will keep you posted

**Comment by kacole2** [comment]
@jvanns `.deb` should be good to go now as well. Please close the issue if you have any more problems

**Comment by jvanns** [comment]
Hi. I've installed this build on a new machine with a minimal configuration that used to work. However, it fails to start as a service;

rexray service -l debug -c /etc/rexray/config.yml start

I've attached the config, stderr and the verbose log all with our AWS access/secret keys removed. 
I can't see anything useful in the error log.

[rexray.yml.txt](https://github.com/emccode/rexray/files/336891/rexray.yml.txt)
[rexray.stderr.txt](https://github.com/emccode/rexray/files/336892/rexray.stderr.txt)
[rexray.log.txt](https://github.com/emccode/rexray/files/336893/rexray.log.txt)

**Comment by kacole2** [comment]
@jvanns what's your OS/AMI? And you are using v0.3.3, correct?

There were changes made to the install script yesterday.

**Comment by jvanns** [comment]
Ubuntu 14.04 AMI.

This is the first time I've attempted to start the service following the new downgrade via the solution above.

**Comment by kacole2** [comment]
@jvanns I just tested with the following and had no problems.

Ubuntu 14.04 uses upstart vs systemd and it's worth noting that Docker Machine favors Ubuntu 15.10 for all AWS deployments now. However, I was able to find an AMI using 14.04 with HVM. I use US-EAST-1 [Amazon EC2 AMI Locator](https://cloud-images.ubuntu.com/locator/ec2/)

```
docker-machine create --driver amazonec2 --amazonec2-access-key my_key --amazonec2-secret-key my_secret_key --amazonec2-vpc-id my_vpc_since_i_use_b --amazonec2-zone "b" --amazonec2-ami ami-3bda1956 aws1404
```

install REX-Ray
```
docker-machine ssh aws1404 "curl -sSL https://dl.bintray.com/emccode/rexray/install | sh -s stable 0.3.3"
```

add the configuration file
```
docker-machine ssh aws1404 "sudo tee -a /etc/rexray/config.yml << EOF
rexray:
  storageDrivers:
  - ec2
aws:
  accessKey: mykey
  secretKey: mysecret key
EOF"
```

start the service
```
docker-machine ssh aws1404 "sudo rexray service start"
```

see a list of volumes
```
docker-machine ssh aws1404 "rexray volume get"
```

**Comment by jvanns** [comment]
Can you make any sense of what's failing in that error log then? I'm finding it difficult to actually see what is failing let alone why.

msg="default module(s) failed to start" 

Any pointers?

**Comment by kacole2** [comment]
@jvanns here is what a successful log should look like:

```
time="2016-06-28T14:08:40Z" level=debug msg="core get drivers" osDrivers=[linux] storageDrivers=[ec2] volumeDrivers=[docker]
time="2016-06-28T14:08:40Z" level=info msg="storage driver initialized" accessKey=******** moduleName=default-docker provider=ec2 secretKey="******"
time="2016-06-28T14:08:40Z" level=info msg="docker volume driver initialized" availabilityZone= iops= moduleName=default-docker provider=docker size= volumeRootPath="/data" volumeType=
time="2016-06-28T14:08:40Z" level=info msg="os driver initialized" moduleName=default-docker provider=linux
time="2016-06-28T14:08:40Z" level=debug msg="checking volume path cache setting" pathCache=true
time="2016-06-28T14:08:40Z" level=info msg=vdm.List driverName=docker moduleName=default-docker
time="2016-06-28T14:08:40Z" level=info msg="listing volumes" driverName=docker moduleName=default-docker
time="2016-06-28T14:08:40Z" level=info msg=sdm.GetVolume driverName=ec2 moduleName=default-docker volumeID= volumeName=
time="2016-06-28T14:08:40Z" level=info msg=odm.GetMounts deviceName= driverName=linux moduleName=default-docker mountPoint=
time="2016-06-28T14:08:40Z" level=debug msg="docker voldriver spec file" path="/etc/docker/plugins/rexray.spec"
time="2016-06-28T14:08:40Z" level=info msg="started module" address="unix:///run/docker/plugins/rexray.sock" name=default-docker typeName=docker
time="2016-06-28T14:08:40Z" level=info msg="service sent registered modules start signals"
time="2016-06-28T14:08:40Z" level=info msg="service successfully initialized, waiting on stop signal"
```

Specifically, there is an issue with `GET MOUNTS` for your OS driver:
```
level=info msg=odm.GetMounts deviceName= driverName=linux moduleName=default-docker mountPoint=
```

Is there a specific public AMI you are using that I can test against?

**Comment by clintonskitson** [comment]
It is possible that the startup is violating the 3 second timeout window in 0.3.3 because of the initial volume lookup that occurs. This happens with the pathcache feature enabled. You could try disabling this feature for now to skip this lookup on startup. The 0.4x release includes a timeout setting and will be relevant shortly when we add back the EC2 driver.

**Comment by byrnedo** [comment]
Just heads up, I got 0.3.3 starting fine on systemd by putting the following in `/etc/systemd/system/rexray.service.d/override.conf`
```
[Service]
Restart=always
StartLimitIntervalSec=100
StartLimitBurst=10
```

**Comment by kacole2** [comment]
@jvanns any updates here? if this is solved, please close it out. thanks!
